Tests/Packet++Test/Tests/DoIpTests.cpp

```diff
@@ -0,0 +1,1402 @@
+	PTF_ASSERT_TRUE(RoutingActivationRequest.isPacketOfType(pcpp::IPv4));
+	PTF_ASSERT_TRUE(RoutingActivationRequest.isPacketOfType(pcpp::TCP));
+	PTF_ASSERT_TRUE(RoutingActivationRequest.isPacketOfType(pcpp::DOIP));
+
+	pcpp::TcpLayer* tcpLayer = RoutingActivationRequest.getLayerOfType<pcpp::TcpLayer>();
+	PTF_ASSERT_NOT_NULL(tcpLayer);
+
+	PTF_ASSERT_EQUAL(tcpLayer->getDstPort(), pcpp::DoIpPorts::TCP_UDP_PORT);
+
+	auto* doipLayer = RoutingActivationRequest.getLayerOfType<pcpp::DoIpRoutingActivationRequest>();
+
+	PTF_ASSERT_NOT_NULL(doipLayer);
+
+	PTF_ASSERT_EQUAL(doipLayer->getProtocolVersion(), pcpp::DoIpProtocolVersion::ISO13400_2012, enumclass);
+	PTF_ASSERT_EQUAL(doipLayer->getInvertProtocolVersion(), 0xFD);
+	PTF_ASSERT_EQUAL(doipLayer->getPayloadType(), pcpp::DoIpPayloadTypes::ROUTING_ACTIVATION_REQUEST, enumclass);
+	PTF_ASSERT_EQUAL(doipLayer->getPayloadTypeAsStr(), "Routing activation request");
+	PTF_ASSERT_EQUAL(doipLayer->getPayloadLength(), 11);
+	PTF_ASSERT_EQUAL(doipLayer->toString(), "DoIP Layer, Routing activation request (0x0005)");
+	PTF_ASSERT_EQUAL(doipLayer->getSourceAddress(), 0x0e80);
+	PTF_ASSERT_EQUAL(doipLayer->getActivationType(), pcpp::DoIpActivationTypes::DEFAULT, enumclass);
+	std::array<uint8_t, 4> isoField = { 0x0, 0x0, 0x0, 0x0 };
+	PTF_ASSERT_VECTORS_EQUAL(doipLayer->getReservedIso(), isoField);
+	PTF_ASSERT_TRUE(doipLayer->hasReservedOem());
+	std::array<uint8_t, 4> oemField = { 0x0, 0x0, 0x0, 0x0 };
+	PTF_ASSERT_BUF_COMPARE(doipLayer->getReservedOem().data(), oemField.data(), 4);
+
+	PTF_ASSERT_EQUAL(
+	    doipLayer->getSummary(),
+	    "Source Address: 0xe80\nActivation type: Default (0x0)\nReserved by ISO: 00000000\nReserved by OEM: 00000000\n");
+}
+
+PTF_TEST_CASE(DoIpRoutingActivationRequestPacketCreation)
+{
+	pcpp::Packet doIpPacket(100);
+	pcpp::EthLayer ethLayer(pcpp::MacAddress("00:13:72:25:fa:cd"), pcpp::MacAddress("00:e0:b1:49:39:02"));
+	pcpp::IPv4Layer ipLayer(pcpp::IPv4Address("172.22.178.234"), pcpp::IPv4Address("10.10.8.240"));
+	pcpp::TcpLayer tcpLayer((uint16_t)13400, (uint16_t)13400);
+
+	tcpLayer.getTcpHeader()->windowSize = 64240;
+	ipLayer.getIPv4Header()->timeToLive = 128;
+
+	PTF_ASSERT_TRUE(doIpPacket.addLayer(&ethLayer));
+	PTF_ASSERT_TRUE(doIpPacket.addLayer(&ipLayer));
+	PTF_ASSERT_TRUE(doIpPacket.addLayer(&tcpLayer));
+	doIpPacket.computeCalculateFields();
+
+	unsigned char bytes[] = { 0x2, 0xfd, 0x0, 0x5, 0x0, 0x0,  0x0,  0xb,  0xe, 0x80,
+		                      0x0, 0x1,  0x2, 0x3, 0x4, 0x05, 0x05, 0x05, 0x05 };
+	unsigned char bytesWithoutOem[] = { 0x2, 0xfd, 0x0, 0x5, 0x0, 0x0, 0x0, 0x7, 0xe, 0x80, 0x0, 0x1, 0x2, 0x3, 0x4 };
+	std::array<uint8_t, 4> isoReserved{ 0x1, 0x2, 0x3, 0x4 };
+	std::array<uint8_t, 4> oemField{ 0x5, 0x5, 0x5, 0x5 };
+
+	pcpp::DoIpRoutingActivationRequest doipLayer(0x00, pcpp::DoIpActivationTypes::WWH_OBD);
+
+	doipLayer.setSourceAddress(0x0e80);
+	doipLayer.setActivationType(pcpp::DoIpActivationTypes::DEFAULT);
```

Comment:
I don't think we need to set `sourceAddress` and `activateType` both in the c'tor and again here. We can set the expected values in the c'tor, and test the setters later in this test.

Same in the other tests below